src/EventCalendar.js

```diff
@@ -65,120 +104,526 @@ class EventCalendar {
+                newPattern = _self.clndr.options.selectedDate.recur().every(1).months();
+                break;
+            case 'annually':
+                newPattern = _self.clndr.options.selectedDate.recur().every(1).year();
+                break;
+            default:
+                newPattern = null;
+                break;
+        }
+
+        _self.recurPattern = newPattern;
     }
 
-    styleToDel (clndr, target) {
-        let $elem = clndr.element.find('[data-day="' + target.date.date._i + '"]');
-        
-        $elem.parent().addClass('event-del');
+    /**
+     * Calculates the dates within the currently viewed month which should be painted by the recurrence pattern, and paints
+     * or unpaints them according to the `method`. The pattern will be retrieved from the clndr instance.
+     * 
+     * @param {string} method `repeat-on` will paint the dates, `clear` will unpaint them
+     */
+    paintRepeatPattern (method) {    
+        let _self = this,
+            paintMethod = (method) ? method : 'repeat-on',
+            recurDatesThisMonth = _self.recurPattern
+                                    .startDate(_self.clndr.options.selectedDate)
+                                    .endDate(_self.clndr.options.constraints.endDate)
+                                    .all();
+
+        $.each(recurDatesThisMonth, function() {
+            switch (paintMethod) {
+                case 'repeat-on':
+                    _self.styleRepeatOn(this);
+                    break;
+                case 'clear':
+                default:
+                    _self.styleClear(this);
+                    break;
+            }
+        });
     }
 
+    /**
+     * Triggered when navigating through or making any changes to the calendar. Takes the dates stored in the 
+     * `datesToAdd` and `datesToDel` arrays as well as the currently stored recurrence pattern and paints the dates
+     * appropriately. Also triggers the updating of the review panel when finished.
+     * 
+     * @param {moment} month A moment object representing the currently viewed month
+     * @param {string} method `repeat-on` will paint the dates, `clear` will unpaint them
+     */
+    paintMonth (month, method) {
+        let _self = this;
+
+        // Unpaint the entire month, used to catch any dates resetting to neutral
+        _self.styleClearAll();
+
+        // Recurrence pattern to paint for the displayed month
+        if (_self.recurPattern != null) {
+            _self.paintRepeatPattern(method);
+        }
+
+        // Exceptions to paint as to-add
+        if (_self.clndr.options.extras.datesToAdd.length) {
+            const datesToAdd = _self.clndr.options.extras.datesToAdd
+                                .filter(EventCalendar.isInYear.bind(this, month))
+                                .filter(EventCalendar.isInMonth.bind(this, month));
+
+            $.each(datesToAdd, function() {
+                _self.styleToAdd(this);
+            });
```

Comment:
Whilst there's nothing wrong with `$.each`, it'd be nice to start bringing in `forEach` (as we have with `filter`)

The refactor here would be:

```
// Column A
datesToAdd.forEach((date) => {
    _self.styleToAdd(date);
})

// Column B
datesToAdd.forEach(date => _self.styleToAdd(date));


// Column Neckbeard
datesToAdd.forEach(_self.styleToAdd.bind(_self));
```

Reply:
implemented neckbeard